tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,234 @@
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.packagemanifest import PackageManifest
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import (
+    get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    Few checks to assert for before and after upgrade
+    1. Checks for pods in openshift-logging project
+    2. Checks for the health of logging cluster before upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+@retry(CommandFailed, 5, 30, 2)
+def get_csv_version(channel):
+    """
+    Get the cluster-logging and Elasticsearch CSV version
+    and Images of the pods
+
+    Args:
+        channel (int) : Logging Channel
+
+    Returns:
+        tuple: Tuple containing three elements
+            cluster_logging_csv (str) : Name of the cluster-logging CSV
+            elasticsearch_csv (str) : Name of the elasticsearch CSV
+            dict: Images dict like: {'image_name': 'image.url.to:tag', ...}
+
+    """
+
+    clo_package_manifest = PackageManifest(
+        resource_name='cluster-logging'
+    )
+    cluster_logging_csv = clo_package_manifest.get_current_csv(channel)
+    es_package_manifest = PackageManifest(
+        resource_name='elasticsearch-operator'
+    )
+    elasticsearch_csv = es_package_manifest.get_current_csv(channel)
+    logging_csv = CSV(
+        resource_name=cluster_logging_csv,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    images = ocp.get_images(logging_csv.get())
+    return cluster_logging_csv, elasticsearch_csv, images
+
+
+def check_csv_version_post_upgrade(channel):
+    """
+    This function is to check the CSV version post upgrade
+
+    Args:
+        channel (int) : logging channel
+
+    Returns:
+        bool : Returns True if version matches, False otherwise
+
+    """
+
+    cluster_logging_csv, elasticsearch_csv, images = get_csv_version(channel)
+    if (
+        config.UPGRADE['upgrade_logging_channel'] in cluster_logging_csv
+        and config.UPGRADE['upgrade_logging_channel'] in elasticsearch_csv
+    ):
+        logger.info(
+            f"Upgraded version of Cluster-logging "
+            f"operator {cluster_logging_csv}"
+        )
+        logger.info(
+            f"Upgraded version of Elastic-search "
+            f"operator {elasticsearch_csv}"
+        )
+        return True
+    return False
+
+
+@retry(AttributeError, 8, 30, 2)
```

Comment:
number 2 -  backoff: backoff multiplier e.g. value of 2 will double the delay each retry

Which will leads to about 2 hours - is this intention here?

The problem I see that if it will fail after delay will be increased to 3840 seconds it's a lot of time of waiting and doing nothing.

I would rather have backoff set to 1, and extend delay to some reasonable time to be able to check once a let say 1 or few minutes. It can save a lot of time and the calculation of maximum wait time will be more straight forward. 

For example 90 min and retry each 60 seconds = 
`@retry(AttributeError, 90, 60, 1)` .